test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSourcesCommandTest.cs

```diff
@@ -416,6 +416,173 @@ public void SourcesCommandTest_AddWithUserNamePassword_UserDefinedConfigFile()
+        {
+            using (SimpleTestPathContext pathContext = new SimpleTestPathContext())
+            {
+                TestDirectory workingPath = pathContext.WorkingDirectory;
+                SimpleTestSettingsContext settings = pathContext.Settings;
+
+                // Arrange
+                string nugetexe = Util.GetNuGetExePath();
+                var args = new string[] {
+                    "sources",
+                    "Add",
+                    "-Name",
+                    "test_source",
+                    "-Source",
+                    "https://source.test",
+                    "-ConfigFile",
+                    settings.ConfigPath,
+                    "-ProtocolVersion",
+                    protocolVersion
+                };
+
+                // Act
+                CommandRunnerResult result = CommandRunner.Run(nugetexe, workingPath, string.Join(" ", args), true);
+
+                // Assert
+                Assert.Equal(shouldSucceed, result.ExitCode.Equals(0));
+            }
+        }
+
+        [Fact]
+        public void SourcesCommandTest_UpdateWithProtocolVersion()
+        {
+            using (TestDirectory configFileDirectory = TestDirectory.Create())
+            {
+                string nugetexe = Util.GetNuGetExePath();
+                string configFileName = "nuget.config";
+                string configFilePath = Path.Combine(configFileDirectory, configFileName);
+
+                var nugetConfig = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <add key=""test_source"" value=""https://source.test.initial"" />
+  </packageSources>
+</configuration>";
+                Util.CreateFile(configFileDirectory, configFileName, nugetConfig);
+
+                // Arrange
+                var args = new string[] {
+                    "sources",
+                    "Update",
+                    "-Name",
+                    "test_source",
+                    "-Source",
+                    "https://source.test",
+                    "-ConfigFile",
+                    configFilePath,
+                    "-ProtocolVersion",
+                    "3"
+                };
+
+                // Act
+                CommandRunnerResult result = CommandRunner.Run(
+                    nugetexe,
+                    configFileDirectory,
+                    string.Join(" ", args),
+                    true);
+
+                // Assert
+                Assert.Equal(0, result.ExitCode);
+                ISettings loadedSettings = Configuration.Settings.LoadDefaultSettings(configFileDirectory, configFileName, null);
+                SettingSection packageSourcesSection = loadedSettings.GetSection("packageSources");
+                Assert.Single(packageSourcesSection.Items);
+                SourceItem sourceItem = packageSourcesSection?.GetFirstItemWithAttribute<SourceItem>("key", "test_source");
+                Assert.Equal("3", sourceItem.ProtocolVersion);
```

Comment:
Please assert logged error message too.

Reply:
Added tests validating unhappy path and error message.